python/pyspark/sql/connect/plan.py

```diff
@@ -575,7 +575,14 @@ def plan(self, session: Optional["RemoteSparkSession"]) -> proto.Relation:
         rel = proto.Relation()
         rel.join.left.CopyFrom(self.left.plan(session))
         rel.join.right.CopyFrom(self.right.plan(session))
-        rel.join.join_condition.CopyFrom(self.to_attr_or_expression(self.on, session))
+        if self.on is not None:
+            if not isinstance(self.on, list):
+                if isinstance(self.on, str):
```

Comment:
is this consistent with the "regular" PySpark approach?

Reply:
If you check existing PySpark implementation, yes, this is what I learn from there to support such multiple types.